### PR TITLE
Doc updates, rearrange rtd website, add user-guides and contributing tabs

### LINKED ISSUE #733: Consider restructuring the documentation page slightly so it's easier to navigate and examples are more evident / "front and center"
Feedback from multiple sources suggest that documentation can be improved both in content and in structure. See this [Hera discussion](https://github.com/argoproj-labs/hera/discussions/625#discussioncomment-6602713) and this [CNCF Slack conversation](https://cloud-native.slack.com/archives/C03NRMD9KPY/p1690278658761069). In order to facilitate onboarding and usage, we should restructure and document parts of Hera's features more

### PR DESCRIPTION
**Pull request by elliotgunton**
* Add a workflows-examples symlink to fix the link for readme and match website
* Make the text in workflows-examples more obvious to fix #733 
* Make CONTRIBUTING its own page on the mkdocs website as it was not reachable except through the readme link, edit text to encourage contributors
* Move expr.md into a user-guides section
* Add core-concepts section outline to user-guides section

### CODE REVIEW

**Comment by elliotgunton** [comment]
Pls don't merge until #755 is merged 🙏

**Comment by flaviuvadan** [comment]
🚀 thanks for addressing the issue!!

**Review by flaviuvadan — APPROVED** [review]

**Pull request merged**
